deep_qa/training/models.py

```diff
@@ -146,6 +151,183 @@ def _make_predict_function(self):
             self.predict_function = Step(inputs, self.outputs, self.global_step,
                                          updates=self.state_updates)
 
+    @overrides
+    def _fit_loop(self,
+                  f: callable,
+                  ins: List[numpy.array],
+                  out_labels: List[str]=None,
+                  batch_size: int=32,
+                  epochs: int=100,
+                  verbose: int=1,
+                  callbacks: List[Callback]=None,
+                  val_f: callable=None,
+                  val_ins: List[numpy.array]=None,
+                  shuffle: bool=True,
+                  callback_metrics: List[str]=None,
+                  initial_epoch: int=0):
+        """
+        Abstract fit function which preprocesses and batches
+        data before training a model. We override this keras backend
+        function to support multi-gpu training via splitting a large
+        batch size across multiple gpus. This function is broadly the
+        same as the Keras backend version aside from this - changed elements
+        have corresponding comments attached.
+
+        Note that this should not be called directly - it is used by calling
+        model.fit()
```

Comment:
No - this is the next step, which would get it working with `DataGenerators`.  I think it should be do-able, possibly requiring overriding large methods in keras' backend only to change small things, but for that reason I was going to leave it for a separate PR.